components/raftstore-v2/src/operation/life.rs

```diff
@@ -570,12 +570,13 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
     /// tablet.
     #[inline]
     pub fn postponed_destroy(&self) -> bool {
+        let last_applying_index = self.compact_log_context().last_applying_index();
         let entry_storage = self.storage().entry_storage();
         // If it's marked as tombstone, then it must be changed by conf change. In
         // this case, all following entries are skipped so applied_index never equals
         // to commit_index.
         (self.storage().region_state().get_state() != PeerState::Tombstone
-           && entry_storage.applied_index() != entry_storage.commit_index())
+            && entry_storage.applied_index() != last_applying_index)
```

Comment:
The comment should be changed to match the change.

Also could you elaborate the difference of entry_storage.commit_index() and last_applying_index?
So the compact_log_context's last_applying_index comes from committed_entries. So it's kind of observed commit index, which should be <= entry_storage.commit_index().
But I still don't get why this change is needed.

Reply:
Because last_applying_index is the largest index of entries sent to apply thread, so applied_index always <= last_applying_index. And applied_index may never equals commit_index.